42/README.md

```diff
@@ -70,14 +71,129 @@ The above process ensure the node keeps validating and accepting new blocks whil
-
-Following describes how to archive and start a new blockchain,
-- Create a TEP which proposes a block height `h` at which the current blockchain will stop creating any new blocks.
-- Once the TEP concludes and the height `h` has been chosen, validators stop proposing new blocks once height `h` has been reached.
-- At height `h` the merkle root of the UTXO of the current blockchain is provided as the genesis state of the new blockchain.
-- The validator set at height `h` should be used as the validator set in the genesis of the new blockchain.
-- Since from Tendermint's point of view this is a new blockchain, block height will start from `0` which implies that BigchainDB would need to archive the existing blocks collection. All other collections would remain as is.
+
+This section advises on how to approach backwards-incompatible upgrades of the Tendermint chain - the type of updgrades when Tendermint can not build blocks on top of the existing chain.
+
+BigchainDB operators need a convenient way to do the following:
+
+1. Stop building blocks when the old Tendermint chain is at a jointly chosen height.
+2. Continue building blocks using the new Tendermint chain starting from that height.
+3. Replay the whole blockchain after a migration.
+4. Join the network as a new validator after a migration.
+
+Additionally, BigchainDB HTTP API usage has to be seamless - all the HTTP responses has to look like there were no migrations.
+
+Below we describe how to perform a migration while meeting the postulated requirements. Note that there is no limitation on the number of consequent migrations that can be performed this way.
+
+#### 1. Stop building blocks
+
+A CLI command can be used to instruct the node to stop building blocks at a chosen height.
+
+To make sure no blocks are committed after the agreed height, we propose an election process:
+
+- The initiator creates an election, which includes a block height.
+- The validators vote for the election. Once the election is concluded, new blocks are not committed by the validators. New transactions from the users are rejected. New blocks sent by Tendermint are rejected too. The conditions for resuming the chain operation are described below.
+
+To perform a migration election, we propose 2 CLI commands.
+
+The initiator executes:
+
+```
+$ bigchaindb migration new <height> --private-key /home/user/.tendermint/config/priv_validator.json
+```
+
+The command outputs the migration ID. The initiator distributes it among other members of the network. The process is similar to adding new validators.
+
+Validators vote for the migration:
+
+```
+$ bigchaindb migration approve <migration-id> --private-key /home/user/.tendermint/config/priv_validator.json
+```
+
+Validators can watch how the election goes:
+
+```
+$ bigchaindb migration status <migration-id>
+```
+
+Outputs:
+```
+votes_recieved=<Sum_of_votes_recieved>
+votes_allocated=<Sum_of_votes_allocated_in_election>
+network_size=<Total_network_power>
+```
```

Comment:
I think it is possible to definitely state that the migration concluded or not. We can just state if the migration has concluded or not.

Reply:
Makes sense. Here I wanted it to look similar to how it looks for `upsert-validator`. We can add an additional `migration_concluded=` line for both, what do you think?

Reply:
Yes :+1:

Reply:
Thinking about conclusion (also related to [this comment](https://github.com/bigchaindb/BEPs/pull/68#discussion_r210209288)), we need to display election status, one of `started`, `concluded`, `failed`.

Let's agree on the names for these statuses and document them here and in [BEP-21](https://github.com/bigchaindb/BEPs/tree/master/21).

Reply:
Instead of `started` I would suggest `ongoing`? i.e. once an election transaction has been committed its status could be described as ongoing

Reply:
Ok, I will add an `election_status=` line but there is no need to enumerate the statuses here. You can just do it in BEP-21.